packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,246 @@
+        <Primitive
+          ref={forwardedRef}
+          as={TOOLBAR_DEFAULT_TAG}
+          selector={getSelector(TOOLBAR_NAME)}
+          role="toolbar"
+          aria-labelledby={ariaLabel ? undefined : ariaLabelledBy}
+          aria-label={ariaLabel || undefined}
+          aria-controls={ariaControls}
+          aria-orientation={orientation}
+          data-orientation={orientation}
+          {...ToolbarProps}
+        >
+          {children}
+        </Primitive>
+      </RovingFocusGroup>
+    </ToolbarContext.Provider>
+  ) as any;
+}) as ToolbarPrimitive;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarGroup
+ * -----------------------------------------------------------------------------------------------*/
+
+const GROUP_NAME = 'ToolbarGroup';
+
+type ToolbarGroupOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarGroupPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarGroupOwnProps
+>;
+
+const ToolbarGroup = React.forwardRef((props, forwardedRef) => {
+  const { orientation } = useToolbarContext(GROUP_NAME);
+
+  return (
+    <Primitive
+      role="group"
+      selector={getSelector(GROUP_NAME)}
+      data-orientation={orientation}
+      {...props}
+      ref={forwardedRef}
+    />
+  );
+}) as ToolbarGroupPrimitive;
+
+ToolbarGroup.displayName = GROUP_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarLabel
+ * -----------------------------------------------------------------------------------------------*/
+
+const LABEL_NAME = 'ToolbarLabel';
+
+type ToolbarLabelOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarLabelPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarLabelOwnProps
+>;
+
+const ToolbarLabel = React.forwardRef((props, forwardedRef) => (
+  <Primitive selector={getSelector(LABEL_NAME)} {...props} ref={forwardedRef} />
+)) as ToolbarLabelPrimitive;
+
+ToolbarLabel.displayName = LABEL_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarSeparator
+ * -----------------------------------------------------------------------------------------------*/
+
+const SEPARATOR_NAME = 'ToolbarSeparator';
+
+type ToolbarSeparatorOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarSeparatorPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarSeparatorOwnProps
+>;
+
+const ToolbarSeparator = React.forwardRef((props, forwardedRef) => {
+  const { orientation } = useToolbarContext(GROUP_NAME);
+
+  return (
+    <Primitive
+      role="separator"
+      selector={getSelector(SEPARATOR_NAME)}
+      aria-orientation={orientation}
+      data-orientation={orientation}
+      {...props}
+      ref={forwardedRef}
+    />
```

Comment:
Sure, but this conflicts with the `Separator` that I am exporting from `Toolbar.tsx` which I quite honestly don't understand the advantage of this pattern, so I imported `Separator` as `SeparatorImpl` for now.

Reply:
Haha, we export `ToolbarSeparator` *and* `Separator` to give our consumers the option of different import styles:

```jsx
import { Toolbar, ToolbarSeparator } from '@radix-ui/react-toolbar';

render(
	<Toolbar>
		<ToolbarSeparator />
	</Toolbar>
);

// or 

import * as Toolbar from '@radix-ui/react-toolbar';

render(
	<Toolbar.Root>
		<Toolbar.Separator />
	</Toolbar.Root>
);
```

Some people prefer dot notation, some don't and it was easy enough for us to support both this way while still allowing tree-shaking to do its thing if they're builds are set up for that.

@benoitgrelard What do you think about the `import { Separator as SeparatorImpl }` in this case? I'm okay with it tbh but interested in whether you have an opinion here.

Reply:
I'd rather just `import { Separator } from …` but because we clash because we can't do that `export ToolbarSeparator as Separator` (because of a parcel bug) the next best thing for me would be to do:

```jsx
import { Separator as SeparatorPrimitive } from '@radix-ui/react-separator'
//or
import * as Separator from '@radix-ui/react-separator'
// and use `Separator.Root
```